player.cpp

```diff
@@ -584,26 +590,45 @@ void test_player() //!OCLINT tests may be long
     // A players goes ?right/?up upon acceleraton
     {
         player p_forward;
+        coordinate c_before = get_coordinate(p_forward);
         p_forward.accelerate();
-        assert(get_dx(p_forward) > 0.0); // Get the delta x, maybe needs to be added
-        assert(get_dy(p_forward) > 0.0); // Get the delta x, maybe needs to be added
+        coordinate c_after = get_coordinate(p_forward);
+        assert(p_forward.get_direction() > -0.00000000001 && p_forward.get_direction() < 0.00000000001);
+        double dx = get_x(c_after) - get_x(c_before);
+        double dy = get_y(c_after) - get_y(c_before);
+        assert(dx > 0.0);
+        assert(dy > -0.00000000001 && dy < 0.00000000001);
+
         player p_backward;
+        c_before = get_coordinate(p_backward);
         p_backward.acc_backward();
-        assert(get_dx(p_backward) < 0.0); // Signs should flip
-        assert(get_dy(p_backward) < 0.0); // Signs should flip
+        c_after = get_coordinate(p_backward);
+        assert(p_backward.get_direction() > -0.00000000001 && p_backward.get_direction() < 0.00000000001);
+        double dx = get_x(c_after) - get_x(c_before);
+        double dy = get_y(c_after) - get_y(c_before);
+        assert(dx < 0.0);
+        assert(dy > -0.00000000001 && dy < 0.00000000001);
     }
     // RJCB: another suggested test
     // A players actually goes backwards after some backwards accelerations
     {
         player p;
+        coordinate c_before = get_coordinate(p);
         p.accelerate();
-        assert(get_dx(p) > 0.0); // Get the delta x, maybe needs to be added
-        assert(get_dy(p) > 0.0); // Get the delta x, maybe needs to be added
+        coordinate c_inbetween = get_coordinate(p);
+        double dx_a = get_x(c_inbetween) - get_x(c_before);
+        double dy_a = get_y(c_inbetween) - get_y(c_before);
+        assert(dx_a > 0.0); // Get the delta x
+        assert(dy_a > -0.00000000001 && dy_a < 0.00000000001);
```

Comment:
That bit is redundant with the block just above isn't it?

Reply:
What do you mean? Pls be more specific :)